### PR TITLE
Add constant & helper library for list resources

### PR DESCRIPTION
**Pull request by karlkfi**

### CODE REVIEW

File: pkg/kinds/lists.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+// IsListGVK returns true if the kind has a "List" suffix.
+func IsListGVK(gvk schema.GroupVersionKind) bool {
+	return strings.HasSuffix(gvk.Kind, ListSuffix)
+}
+
+// ListGVKForItemGVK returns the item GroupVersionKind with "List" appended to
+// the kind.
+func ListGVKForItemGVK(gvk schema.GroupVersionKind) schema.GroupVersionKind {
+	gvk.Kind += ListSuffix
```
Comment (sdowell) [inline pkg/kinds/lists.go:38]: Maybe this should check whether the GVK already has a List suffix to detect misuse, WDYT?

**Comment by karlkfi** [comment]
/retest 

Looks like the new fake.WatchSupervisor has a race condition:

```
I0526 23:07:05.431595   18737 root.go:122] Parsing files from source dir: 
I0526 23:07:05.451713   18737 root.go:122] Parsing files from source dir: 
I0526 23:07:05.506690   18737 root.go:122] Parsing files from source dir: 
I0526 23:07:05.521479   18737 root.go:122] Parsing files from source dir: 
panic: send on closed channel
goroutine 114 [running]:
kpt.dev/configsync/pkg/syncer/syncertest/fake.(*WatchSupervisor).Send(...)
	/go/src/kpt.dev/configsync/pkg/syncer/syncertest/fake/watcher.go:103
kpt.dev/configsync/pkg/syncer/syncertest/fake.(*MemoryStorage).sendPutEvent(0xc00108e040, {0x1fe7df8, 0xc00108eb40}, {{{0xc000115120, 0x11}, {0x189578e, 0x8}}, {{0x1d1ce4f, 0x18}, {0x1d00415, ...}}}, ...)
	/go/src/kpt.dev/configsync/pkg/syncer/syncertest/fake/storage.go:209 +0x245
kpt.dev/configsync/pkg/syncer/syncertest/fake.(*SubresourceStorage).Update(0xc000fdadc8, {0x1fe7df8, 0xc00108eb40}, {0x1ffabb8, 0xc000dafb80}, 0xc0018fa1e0)
	/go/src/kpt.dev/configsync/pkg/syncer/syncertest/fake/subresource_storage.go:129 +0xa3a
kpt.dev/configsync/pkg/syncer/syncertest/fake.(*statusWriter).Update(0xc00[141](https://oss.gprow.dev/view/gs/oss-prow-build-kpt-config-sync/pr-logs/pull/GoogleContainerTools_kpt-config-sync/624/kpt-config-sync-presubmit/1662232804370419712#1:build-log.txt%3A141)8000, {0x1fe7df8, 0xc00108eb40}, {0x1ffabb8, 0xc000dafb80}, {0x0, 0x0, 0xc1[146](https://oss.gprow.dev/view/gs/oss-prow-build-kpt-config-sync/pr-logs/pull/GoogleContainerTools_kpt-config-sync/624/kpt-config-sync-presubmit/1662232804370419712#1:build-log.txt%3A146)cc65f8e73df?})
	/go/src/kpt.dev/configsync/pkg/syncer/syncertest/fake/client.go:162 +0x94
kpt.dev/configsync/pkg/parse.(*root).setSyncStatusWithRetries(0xc000dc0000, {0x1fe7df8, 0xc00108eb40}, {0x1, {0x0, 0x0}, {0x0, 0x0}, {{0xc1146cc65f8e73df, 0x1c06f559, ...}}}, ...)
	/go/src/kpt.dev/configsync/pkg/parse/root.go:411 +0xbbb
kpt.dev/configsync/pkg/parse.(*root).SetSyncStatus(0x2e89f00?, {0x1fe7df8?, 0xc00108eb40?}, {0x1, {0x0, 0x0}, {0x0, 0x0}, {{0xc1146cc65f8e73df, 0x1c06f559, ...}}})
	/go/src/kpt.dev/configsync/pkg/parse/root.go:362 +0x11e
kpt.dev/configsync/pkg/parse.setSyncStatus({0x1fe7df8, 0xc00108eb40}, {0x1fed8a8, 0xc000dc0000}, 0xc0010881a0, 0x1, {0x0?, 0x0?})
	/go/src/kpt.dev/configsync/pkg/parse/run.go:428 +0x1b2
kpt.dev/configsync/pkg/parse.updateSyncStatusPeriodically({0x1fe7df8, 0xc00108eb40}, {0x1fed8a8, 0xc000dc0000}, 0xc000d820c0?)
	/go/src/kpt.dev/configsync/pkg/parse/run.go:467 +0x274
created by kpt.dev/configsync/pkg/parse.parseAndUpdate
	/go/src/kpt.dev/configsync/pkg/parse/run.go:397 +0x570
FAIL	kpt.dev/configsync/pkg/parse	0.513s
```

Will fix in a different PR.

File: pkg/kinds/lists.go (hunk 1 above)
Comment (karlkfi) [reply pkg/kinds/lists.go:38]: I deliberately didn't validate here to make it easier to use without the error return. I think it's up to the caller whether they want to validate or not. Same for removing the suffix. But in general, yes, most places should probably check.

**Review by sdowell — APPROVED** [review]

**Comment by karlkfi** [comment]
Fix for unit test flake: https://github.com/GoogleContainerTools/kpt-config-sync/pull/625

**Pull request merged**